beacon-chain/rpc/prysm/v1alpha1/validator/duties.go

```diff
@@ -113,19 +139,23 @@ func (vs *Server) duties(ctx context.Context, req *ethpb.DutiesRequest) (*ethpb.
 				assignment.Committee = ca.Committee
 				assignment.AttesterSlot = ca.AttesterSlot
 				assignment.CommitteeIndex = ca.CommitteeIndex
+				assignment.CommitteesAtSlot = committeesAtSlot
 			}
 			// Save the next epoch assignments.
 			ca, ok = nextEpochAssignments[idx]
 			if ok {
 				nextAssignment.Committee = ca.Committee
 				nextAssignment.AttesterSlot = ca.AttesterSlot
 				nextAssignment.CommitteeIndex = ca.CommitteeIndex
+				nextAssignment.CommitteesAtSlot = nextCommitteesAtSlot
 			}
 		} else {
-			// If the validator isn't in the beacon state, try finding their deposit to determine their status.
-			// We don't need the lastActiveValidatorFn because we don't use the response in this.
-			vStatus, _ := vs.validatorStatus(ctx, s, pubKey, nil)
-			assignment.Status = vStatus.Status
+			log.WithFields(logrus.Fields{
+				"pubKey": hexutil.Encode(pubKey),
+				"idx":    idx,
+			}).Debug("Could not get validator assignment status")
+			assignment.Status = ethpb.ValidatorStatus_UNKNOWN_STATUS
+			nextAssignment.Status = ethpb.ValidatorStatus_UNKNOWN_STATUS
```

Comment:
i think it was a bug that nextAssignment wasn't set here because it skipped adding the information anyway.